vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java

```diff
@@ -1713,4 +1720,92 @@ private static <T> void textEncodeArray(T[] values, DataType type, ByteBuf buff)
     }
     buff.writeByte('}');
   }
+
+  private static Cidr binaryDecodeCidr(int index, int len, ByteBuf buff){
+    byte family = buff.getByte(index);
+    byte netmask = buff.getByte(index+1);
+    Integer val;
+    int size = buff.getByte(index+3);
+    byte[] data = new byte[size];
+    buff.getBytes(index+4,data);
+    InetAddress address;
+
+    switch (family){
+      case 2:
+        // IPV4
```

Comment:
IPV4 and IPV6 seems to be the same code and could be

if (family == 2 || family == 3) instead

Reply:
Addressed.